aten/src/ATen/native/xnnpack/Factory.cpp

```diff
@@ -8,11 +8,16 @@ namespace native {
 namespace xnnpack {
 namespace internal {
 
+GuardingAllocator<0u, XNN_EXTRA_BYTES>* get_guarding_allocator() {
+  static GuardingAllocator<0u, XNN_EXTRA_BYTES> allocator;
+  return &allocator;
+}
+
 Tensor empty_with_tail_padding(
     const IntArrayRef size,
     const caffe2::TypeMeta dtype,
     const c10::MemoryFormat memory_format) {
-  static GuardingAllocator<0u, XNN_EXTRA_BYTES> allocator;
+  auto* allocator_ptr = get_guarding_allocator();
```

Comment:
Just curious - can we ever arrive here from different threads now that we are using custom op registration?  I am not sure if the allocator is thread safe.

Reply:
Thats a good point. Thanks for pointing it out. But I think this is ok, because allocator is not keeping track of any data that my introduce race?